tasks/hyrax-metadata-updates/index.js

```diff
@@ -0,0 +1,306 @@
+    throw new InvalidArgument('Provider not supplied in configuration. Unable to construct path');
+  }
+  const entryTitle = await getEntryTitle(config, metadata, isUmmG);
+  return `providers/${providerId}/collections/${entryTitle}/granules/${getGranuleUr(metadata, isUmmG)}`;
+}
+
+/**
+ * generateHyraxUrl
+ *
+ * @param {Object} config - the config
+ * @param {Object} metadata - the dom
+ * @param {boolean} isUmmG - UMM-G or ECHO10
+ * @returns {string} - the hyrax url
+ */
+async function generateHyraxUrl(config, metadata, isUmmG) {
+  const path = await generatePath(config, metadata, isUmmG);
+  const url = new URL(`${generateAddress()}/${path}`);
+  return (url.href);
+}
+
+/**
+ * addHyraxUrlToUmmG
+ *
+ * @param {Object} metadata - the metadata dom
+ * @param {string} hyraxUrl - the hyrax url
+ * @returns {string} - the updated metadata containing a Hyrax URL
+ */
+function addHyraxUrlToUmmG(metadata, hyraxUrl) {
+  const metadataCopy = cloneDeep(metadata);
+
+  if (isUndefined(metadataCopy.RelatedUrls)) {
+    metadataCopy.RelatedUrls = [];
+  }
+  const url = {
+    URL: hyraxUrl,
+    Type: 'GET DATA',
+    Subtype: 'OPENDAP DATA',
+    Description: 'OPeNDAP request URL'
+  };
+  metadataCopy.RelatedUrls.push(url);
+
+  return JSON.stringify(metadataCopy, null, 2);
+}
+
+/**
+ * addHyraxUrlToEcho10
+ *
+ * @param {Object} metadata - the metadata dom
+ * @param {string} hyraxUrl - the hyrax url
+ * @returns {string} - the updated metadata containing a Hyrax URL
+ */
+function addHyraxUrlToEcho10(metadata, hyraxUrl) {
+  const metadataCopy = cloneDeep(metadata);
+
+  // xml2js will model a single child as an element rather than a list so
+  // we have to defend against that by reconstructing OnlineResources as
+  // a list from scratch
+  delete metadataCopy.Granule.OnlineResources;
+  metadataCopy.Granule.OnlineResources = {};
+  metadataCopy.Granule.OnlineResources.OnlineResource = [];
+
+  if (!isUndefined(metadata.Granule.OnlineResources)) {
+    const urls = metadata.Granule.OnlineResources.OnlineResource;
+    metadataCopy.Granule.OnlineResources.OnlineResource.push(urls);
+  }
+  const url = {
+    URL: hyraxUrl,
+    Description: 'OPeNDAP request URL',
+    Type: 'GET DATA : OPENDAP DATA'
+  };
+  metadataCopy.Granule.OnlineResources.OnlineResource.push(url);
+  return generateEcho10XMLString(metadataCopy.Granule);
+}
+
+/**
+ * addHyraxUrl
+ *
+ * @param {Object} metadata - the original metadata dom
+ * @param {boolean} isUmmG - UMM-G or ECHO10 metadata
+ * @param {string} hyraxUrl - the hyrax url
+ * @returns {string} - the updated metadata containing a Hyrax URL
+ */
+function addHyraxUrl(metadata, isUmmG, hyraxUrl) {
+  return isUmmG ? addHyraxUrlToUmmG(metadata, hyraxUrl) : addHyraxUrlToEcho10(metadata, hyraxUrl);
+}
+
+/**
+ * createDom
+ *
+ * @param {Object} metadataFileName file name
+ * @param {string} metadata - raw metadata
+ * @returns {Object} document object model and whether it is UMM-G
```

Comment:
Should update this return JSdoc - we're not returning a document object model but a JS object